### PR TITLE
hw6 is completed

### PR DESCRIPTION
**Pull request by Dasha-Belozerceva**

### CODE REVIEW

File: src/main/java/hw6/composite/SidebarComposite.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    private List<WebElement> leftSidebarElements;
+
+    @FindBy(linkText = "Service")
+    private WebElement serviceButton;
+
+    @FindBy(linkText = "Different elements")
+    private WebElement differentElementsButton;
+
+    @FindBy(linkText = "User Table")
+    private WebElement userTableButton;
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw6/composite/SidebarComposite.java]: These items could be combined to the `List<WebElement>`

File: src/test/java/hw6/steps/ThenSteps.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    public LogrowTable logrowTable(Map<String, String> entry){
+        return new LogrowTable(entry.get("value"), entry.get("condition"));
+    }
+
+    @Then("LogRow should contain following values:")
+    public void logrowShouldContain(List<LogrowTable> logrowTables){
+        for(int i = 0; i < logrowTables.size(); i++){
+            assertTrue(differentElementsPage.logRowsElementIsDisplayed(logrowTables.get(i).getLogrowValue(),
+                    logrowTables.get(i).getLogrowCondition()));
+        }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/ThenSteps.java]: May be ti will be better set `logrowTable` as parameter for the `differentElementsPage.logRowsElementIsDisplayed()` method?

File: src/test/java/hw6/steps/ThenSteps.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        return new UserTable(entry.get("Number"), entry.get("User"), entry.get("Description"));
+    }
+
+    @Then("User table should contain following values:")
+    public void userTableShouldContain(List<UserTable> userTableEntity){
+        for (int i = 0; i < userTableEntity.size(); ++i){
+            assertEquals(userTablePage.numbersTexts().get(i), userTableEntity.get(i).getNumber());
+            assertEquals(userTablePage.usernameTexts().get(i), userTableEntity.get(i).getUser());
+            assertEquals(userTablePage.descriptionTexts().get(i), userTableEntity.get(i).getDescription());
+        }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/ThenSteps.java]: This loop could be replaced by `assertEquals(userTablePage.getUserTableData(), userTableEntity);`

File: src/test/java/hw6/steps/ThenSteps.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        return new DropdownTable(entry.get("Dropdown Values"));
+    }
+
+    @Then("droplist should contain values in column Type for user {string}")
+    public void droplistShouldContain(String name, List<DropdownTable> dropdownTable){
+        for (int i = 0; i < dropdownTable.size(); i++){
+            assertEquals(userTablePage.optionsInNumberTypeDropdown(name).get(i),
+                    dropdownTable.get(i).getDropdownValue());
+        }
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/ThenSteps.java]: same story as for `LogRowTable`

File: src/test/java/hw6/steps/ThenSteps.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Then("1 log row has {string} text in log section")
+    public void logrowShouldHasString(String str){
+        assertTrue(userTablePage.isLogRowContainsThatValue(str));
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/ThenSteps.java]: What will happen if we will have 3 row logs with the required value?

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]
Please resolve comments

File: src/test/java/hw6/steps/ThenSteps.java (hunk 2 above)
Comment (Dasha-Belozerceva) [reply src/test/java/hw6/steps/ThenSteps.java]: done

File: src/test/java/hw6/steps/ThenSteps.java (hunk 3 above)
Comment (Dasha-Belozerceva) [reply src/test/java/hw6/steps/ThenSteps.java]: Yes, good idea
Done

File: src/test/java/hw6/steps/ThenSteps.java (hunk 4 above)
Comment (Dasha-Belozerceva) [reply src/test/java/hw6/steps/ThenSteps.java]: done

File: src/main/java/hw6/composite/SidebarComposite.java (hunk 1 above)
Comment (Dasha-Belozerceva) [reply src/main/java/hw6/composite/SidebarComposite.java]: done

File: src/test/java/hw6/steps/ThenSteps.java (hunk 5 above)
Comment (Dasha-Belozerceva) [reply src/test/java/hw6/steps/ThenSteps.java]: done. I check only for row which indicated in feature step

**Review by DmitryKhodakovsky — APPROVED** [review]
Approved

**Pull request merged**